server/utils/cloudFunctions/functions/test/algoliaTests/algoliaIntegrationTests.test.js

```diff
@@ -0,0 +1,242 @@
+  // After every 'describe' block, reset the test environments. 
+  // Since these are live dev DBs, this process can be very flaky.
+  // In general, don't expect an 'advertiser' will be reset from 
+  // test to test. 
+  // Note: this will also delete ad records in the corresponding 
+  // algolia index. 
+  after(() => {
+    deleteCollection(DB, 'dev_ads');
+  });
+
+  describe("test_createRecord", () => {
+    it("should propogate an ad document into algolia index", function(done) {
+      const ad = {advertiser: "a", startDate: "2019-10-15"};
+
+      DEV_ADS_COLLECTION.doc("a").set(ad).then(() => {
+        setTimeout(function(){
+          DEV_ADS_INDEX.getObject("a").then(content => {
+            chai.expect(content.data).to.deep.equal(ad);
+          }).catch(err => console.log(err));
+
+          done();
+        }, TIMEOUT_10S);
+      });
+    }).timeout(TIMEOUT_15S);
+    
+    it("should propogate multiple ad documents without overwriting", function(done) {
+      const adList = [{advertiser: "b", startDate: "2019-10-15"},
+                      {advertiser: "c", startDate: "2019-10-15"},
+                      {advertiser: "d", startDate: "2019-10-16"}];
+
+      Promise.allSettled([DEV_ADS_COLLECTION.doc("b").set(adList[0]),
+                          DEV_ADS_COLLECTION.doc("c").set(adList[1]),
+                          DEV_ADS_COLLECTION.doc("d").set(adList[2])]).then(() => {
+        setTimeout(function(){
+          DEV_ADS_INDEX.getObjects(["b", "c", "d"]).then(content => {
+            chai.expect(content.results[0].data).to.deep.equal(adList[0]);
+            chai.expect(content.results[1].data).to.deep.equal(adList[1]);
+            chai.expect(content.results[2].data).to.deep.equal(adList[2]);
+          }).catch(err => console.log(err));
+
+          done();
+        }, TIMEOUT_10S);
+      });
+    }).timeout(TIMEOUT_15S);
+  });
+
+  describe("test_updateRecord", () => {
+    it("should propogate a firestore document update to " +
+       "an algolia record ", function(done) {
+      const ad = {advertiser: "e", startDate: "2019-10-15"};
+
+      DEV_ADS_COLLECTION.doc("e").set(ad).then(() => {
+        DEV_ADS_COLLECTION.doc("e").update({startDate: "2020-10-15"}).then(() => {
+          setTimeout(function() {
+
+            // Verify that `ad` was updated.
+            DEV_ADS_INDEX.getObject("e").then(content => {
+              chai.expect(content.data)
+                  .to.deep.equal({advertiser: "e", startDate: "2020-10-15"});
+            }).catch(err => console.log(err));
+            
+            done(); 
+          }, TIMEOUT_10S);
+        });
+      });
+    }).timeout(TIMEOUT_15S);
+
+    it("should propogate a firestore document update to " + 
+       "the proper algolia record", function(done) {
+      const adOne = {advertiser: "f", startDate: "2019-10-15"};
+      const adTwo = {advertiser: "g", startDate: "2019-10-15"};
+    
+      Promise.allSettled([DEV_ADS_COLLECTION.doc("f").set(adOne),
```

Comment:
This promise chaining gets hard to read.  My eyes keep having to bounce around vertically to align your blocks.   Depending on your stack dumps it may also be harder to trace.
Is it possible to rewrite it as something like:

Promise writeF = DEV_ADS_COLLECTION.doc("f").set(adOne);
Promise writeG = DEV_ADS_COLLECTION.doc("g").set(adOne);
await Promise.allSettled(writeF, writeG);
await DEV_ADS_COLLECTION.doc("f").update({startDate: "2020-10-15"}).timeout(TIMEOUT_15S);
setTimeout(...)